src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(harbivore) {
+    if (harbivore instanceof Herbivore && !harbivore.hidden) {
+      harbivore.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter((animal) => animal.health <= 0);
```

Comment:
we don't really need to run this every time. if we only run the filter if the health goes to 0 or below, it would optimize the code quite a bit in certain scenarios